megamek/src/megamek/common/util/MegaMekFile.java

```diff
@@ -1,43 +1,179 @@
+ *
+ * MekHQ is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * MekHQ is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with MekHQ.  If not, see <http://www.gnu.org/licenses/>.
+ */
 package megamek.common.util;
 
 import java.io.File;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Set;
+import java.util.TreeSet;
 
 import megamek.common.Configuration;
+import megamek.common.logging.DefaultMmLogger;
+import megamek.common.logging.MMLogger;
 
 /**
  * This is a local MegaMek version of java.io.File and is designed to support
  * having files that could exist in two locations: the MM install location and
- * a userdata directory.  When a file is opened, the path is first checked to
- * see if it exists within the userdata directory, and if it does, that file is
+ * a userData directory.  When a file is opened, the path is first checked to
+ * see if it exists within the userData directory, and if it does, that file is
  * opened.  However, if it doesn't exist, then the file is opened from MM's
  * install directory instead.
  * 
  * @author arlith
  *
  */
 public class MegaMekFile {
-    
-    File file;
-    
+    private List<File> fileList = new ArrayList<>();
+    private List<File> subdirectoryList = new ArrayList<>();
+    private boolean isDirectory;
+    private static MMLogger logger = DefaultMmLogger.getInstance();
+
     public MegaMekFile(File parent, String child) {
         this(new File(parent, child).toString());
     }
-    
-    public MegaMekFile(String pathname) {
-        File userdataVersion = new File(Configuration.userdataDir(), pathname);
-        if (userdataVersion.exists()) {
-            file = userdataVersion;
+
+    public MegaMekFile(String pathName) {
+        File standardVersion = new File(pathName);
+        File userDataVersion = new File(Configuration.userdataDir(), pathName);
+        isDirectory = standardVersion.isDirectory();
+
+        if (isDirectory) {
+            // Create an array of files located in the standard file directory
+            List<File> standardFiles = splitFilesAndSubdirectories(standardVersion.listFiles(), true);
+
+            if (userDataVersion.exists()) {
+                // If we have a user data version of the directory, then we need to do some additional
+                // processing based on the files located within it
+                List<File> userDataSubdirectories = splitFilesAndSubdirectories(standardVersion.listFiles(), false);
```

Comment:
This doesn't go but one layer deep, right?

Reply:
Correct. Anything deeper should call this again

Reply:
I'll put more thought into use cases to determine if this is the best way forward or if it would be better to immediately handle subdirectories.

Reply:
I've decided that subdirectories should be handled here too, to clean up code elsewhere.